File: claude-cred.js
/**
 * claude-cred.js - Credential resolution for Claude Code Node.js MCP servers.
 *
 * Usage:
 *   const path = require('path');
 *   const os = require('os');
 *   const { resolve, loadEnvFile } = require(
 *     path.join(os.homedir(), '.claude/super-manager/credentials/claude-cred.js')
 *   );
 *   loadEnvFile(__dirname + '/.env');
 */

const { execSync } = require('child_process');
const fs = require('fs');
const path = require('path');

const SERVICE = 'claude-code';

/**
 * Resolve a credential from the OS credential store.
 * @param {string} key - Credential key in 'service/VARIABLE' format
 * @returns {string} The credential value
 * @throws {Error} If credential not found
 */
function resolve(key) {
    try {
        // Shell out to Python keyring - only happens at startup, not per-request
        const cmd = `python -c "import keyring; v=keyring.get_password('${SERVICE}','${key}'); print(v if v else '')"`;
        const result = execSync(cmd, { encoding: 'utf-8', timeout: 10000 }).trim();
        if (!result || result === 'None') {
            throw new Error('Credential not found: ' + key);
        }
        return result;
    } catch (e) {
        throw new Error('Failed to resolve credential ' + key + ': ' + e.message);
    }
}

/**
 * If value starts with 'credential:', resolve from OS store. Otherwise return as-is.
 */
function resolveValue(value) {
    if (typeof value === 'string' && value.startsWith('credential:')) {
        return resolve(value.slice('credential:'.length));
    }
    return value;
}

/**
 * Load .env file and resolve credential: prefixes into process.env.
 * @param {string} envPath - Path to .env file
 * @returns {Object} Resolved key-value pairs
 */
function loadEnvFile(envPath) {
    if (!fs.existsSync(envPath)) return {};
    const result = {};
    const content = fs.readFileSync(envPath, 'utf-8');
    for (const line of content.split('\n')) {
        const trimmed = line.trim();
        if (!trimmed || trimmed.startsWith('#') || !trimmed.includes('=')) continue;
        const eqIdx = trimmed.indexOf('=');
        const key = trimmed.slice(0, eqIdx).trim();
        let value = trimmed.slice(eqIdx + 1).trim().replace(/^["']|["']$/g, '');
        value = resolveValue(value);
        result[key] = value;
        process.env[key] = value;
    }
    return result;
}

module.exports = { resolve, resolveValue, loadEnvFile };

File: claude_cred.py
"""
claude_cred.py - Credential resolution for Claude Code MCP servers and skills.

Usage (replaces manual .env loader):
    import sys, os
    sys.path.insert(0, os.path.expanduser('~/.claude/super-manager/credentials'))
    from claude_cred import load_env
    load_env()

Or resolve a single credential:
    from claude_cred import resolve
    token = resolve("wiki-lite/CONFLUENCE_API_TOKEN")
"""
import os
import inspect
from pathlib import Path

SERVICE = "claude-code"

def resolve(key):
    """Resolve a credential from the OS credential store.

    Args:
        key: Credential key in 'service/VARIABLE' format
    Returns:
        The credential value string.
    Raises:
        ValueError if not found.
    """
    import keyring
    value = keyring.get_password(SERVICE, key)
    if value is None:
        raise ValueError("Credential not found in OS store: {}".format(key))
    return value

def resolve_value(value):
    """If value starts with 'credential:', resolve from OS store. Otherwise return as-is."""
    if isinstance(value, str) and value.startswith("credential:"):
        return resolve(value[len("credential:"):])
    return value

def load_env(env_path=None, set_environ=True):
    """Load .env file and resolve credential: prefixes.

    Args:
        env_path: Path to .env file. If None, auto-detects from calling script's directory.
        set_environ: If True, sets os.environ (default behavior matching existing servers).
    Returns:
        Dict of resolved key-value pairs.
    """
    if env_path is None:
        caller_file = inspect.stack()[1].filename
        env_path = str(Path(caller_file).parent / '.env')

    env_file = Path(env_path)
    result = {}

    if not env_file.exists():
        return result

    for line in env_file.read_text().splitlines():
        line = line.strip()
        if not line or line.startswith('#') or '=' not in line:
            continue
        key, value = line.split('=', 1)
        key = key.strip()
        value = value.strip().strip('"\'')

        # Resolve credential: prefix
        value = resolve_value(value)

        result[key] = value
        if set_environ:
            os.environ[key] = value

    return result
